discovery/gossiper.go

```diff
@@ -3004,8 +3004,26 @@ func (d *AuthenticatedGossiper) handleChanUpdate(nMsg *networkMsg,
 	// broadcast the channel update announcement if it has an attached
 	// authentication proof. We also won't broadcast the update if it
 	// contains an alias because the network would reject this.
+	//
+	// NOTE: We do not check the dont_forward bit in the channel update
+	// msg because without the authentication proof a channel will not be
+	// broadcasted to the broader network anyways.
 	var announcements []networkMsg
 	if chanInfo.AuthProof != nil && !d.cfg.IsAlias(upd.ShortChannelID) {
+		// We log the case where the dont_forward bit is set although
+		// the channel is already announced to the network because we
+		// already have a AuthProof for it.
+		if upd.MessageFlags.HasDontForward() {
+			log.Warnf("Received %v with the dont_forward bit set "+
+				"(msgflags=%s) for channel=%v from peer=%x "+
+				"for an announced channel",
+				upd.MsgType().String(),
+				upd.MessageFlags.String(),
+				upd.ShortChannelID.ToUint64(),
+				remotePubKey,
+			)
+		}
```

Comment:
I was also thinking a lot how we want to classify this situation. I decided to go with a warning to catch the eye of a potential user. But yeah this bit was introduced to give implementations an efficiency boost when they are having a lot of private channels. I think other implementations do not use a specific range for Aliases therefore this bit saves them a table lookup. But for lnd we always check if it is an alias (`!d.cfg.IsAlias(upd.ShortChannelID)`. So this warning would only help surface peer implementation bugs, no real benefit for lnd. 

Let's see what eugene has to say about this.

Reply:
the spec changed to require this bit if:
- the channel doesn't have the `announce_channel` bit set
- the channel is using an alias in the channel update

ziggie is right that this is for other implementations -- all we need to do is set the bit when sending it out so that another implementation doesn't disconnect us. we don't care about whether they set the bit or not because we know whether or not to forward based on the two points above